CMakeLists.txt

```diff
@@ -327,8 +364,20 @@ if(CMAKE_USE_OPENSSL)
     check_include_file("openssl/ssl.h"    HAVE_OPENSSL_SSL_H)
     check_include_file("openssl/x509.h"   HAVE_OPENSSL_X509_H)
     check_include_file("openssl/rand.h"   HAVE_OPENSSL_RAND_H)
-  elseif(WIN32)
-    set(CURL_WINDOWS_SSPI ON)
+    check_symbol_exists(RAND_status "${CURL_INCLUDES}" HAVE_RAND_STATUS)
+    check_symbol_exists(RAND_screen "${CURL_INCLUDES}" HAVE_RAND_SCREEN)
+    check_symbol_exists(RAND_egd    "${CURL_INCLUDES}" HAVE_RAND_EGD)
+  endif()
+endif()
+
+option(CMAKE_USE_MBEDTLS "Enable mbedTLS for SSL/TLS" OFF)
+if(CMAKE_USE_MBEDTLS)
+  find_package(MbedTLS)
+  if(MBEDTLS_FOUND)
```

Comment:
This should fail hard and early when package MbedTLS was not found (like we discussed for CoreFoundation/Security)